### PR TITLE
kvserver: replace ad-hoc stats in RaftTransport by metrics

### PR DESCRIPTION
**Pull request by pav-kv**
These stats were only printed in text to the logs, and hard to use and match
with other existing metrics in monitoring.

There is a queue size metric introduced in 5d51ab34 which properly exports the
`queue` counter to monitoring. The exported graphs allow observing both the
current, and historical queue sizes (including the approx. max size), so this
PR safely removes both stats as obsolete.

This PR also similarly replaces all the remaining ad-hoc counters with proper
metrics, and removes the plumbing and data structures that were previously
maintaining those stats.

The granularity of the new metrics is per-node/transport. This is due to the
metrics infrastructure not being particularly good at handling high cardinality
combinations of labels such as O(N^2) node ID pairs.

Part of #83917

Release note: None

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
@ajwerner This removes your TODO because it's no longer relevant after this PR: we reduce the granularity of counters. Is it fine, or you would still leave/reword the TODO?

**Review by tbg — APPROVED** [review]
Nice!

File: pkg/kv/kvserver/raft_transport_metrics.go [hunk 1]
```diff
@@ -45,5 +47,19 @@ messages to at least one peer.`,
 			Measurement: "Messages",
 			Unit:        metric.Unit_COUNT,
 		}),
+
+		ResponsesSent: metric.NewCounter(metric.Metadata{
+			Name:        "raft.transport.resp-sent",
+			Help:        "Number of Raft responses sent by the Raft Transport",
```
[the remark this answers is not in the document]
Comment (pavelkalinnikov) [reply pkg/kv/kvserver/raft_transport_metrics.go]: Done.

**Comment by pavelkalinnikov** [comment]
bors r=tbg

**Pull request merged**